Annoying warning stemming from the header file

### Perform all steps below and tick them with [x]
- [x] Read the [FAQ](https://docs.lvgl.io/master/intro/index.html#faq)
- [x] Check the related part of the [Documentation](https://docs.lvgl.io/)
- [x] Update lvgl to the latest version
- [x] Reproduce the issue in a [Simulator](https://docs.lvgl.io/latest/en/html/get-started/pc-simulator.html)

### Describe the bug

When using LVGL headers in my code, I get this warning...

```
In file included from ../lvgl/src/core/lv_obj.h:140,
                 from ../lvgl/lvgl.h:34,
                 from ../MyProject/File.hpp:3,
                 from ../MyProject/File.cpp:2:
../lvgl/src/core/lv_obj_style.h: In function 'void lv_obj_remove_style_all(_lv_obj_t*)':
../lvgl/src/core/lv_obj_style.h:94:48: warning: bitwise operation between different enumeration types '<unnamed enum>' and '<unn
amed enum>' is deprecated [-Wdeprecated-enum-enum-conversion]
   94 |     lv_obj_remove_style(obj, NULL, LV_PART_ANY | LV_STATE_ANY);
      |                                    ~~~~~~~~~~~~^~~~~~~~~~~~~~
```

It is a CMake/C++20 code base, which, I guess, is the reason for this warning. While I have suppressed warnings for the LVGL target using CMake, I still get it from all the files that include `lvgl.h`. Therefore, for every file that include `lvgl.h`, a new warning comes up. 

The source of the warning is from the following function:

```c
// lvgl/src/core/lv_obj_style.h:94

static inline void lv_obj_remove_style_all(struct _lv_obj_t * obj)
{
    lv_obj_remove_style(obj, NULL, LV_PART_ANY | LV_STATE_ANY);
}
```

Moving the implementation to a `.c` file would probably fix the issue but I am not sure of the implications. 

### To Reproduce
```c
#include "lvgl.h"
```

--- a/src/core/lv_obj_style.c
+++ b/src/core/lv_obj_style.c
@@ -152,6 +152,11 @@ void lv_obj_remove_style(lv_obj_t * obj, const lv_style_t * style, lv_style_sele
     }
 }
 
+void lv_obj_remove_style_all(struct _lv_obj_t * obj)
+{
+    lv_obj_remove_style(obj, NULL, LV_PART_ANY | LV_STATE_ANY);
+}
+
 void lv_obj_report_style_change(lv_style_t * style)
 {
     if(!style_refr) return;

--- a/src/core/lv_obj_style.h
+++ b/src/core/lv_obj_style.h
@@ -89,10 +89,7 @@ void lv_obj_remove_style(struct _lv_obj_t * obj, const lv_style_t * style, lv_st
  * Remove all styles from an object
  * @param obj       pointer to an object
  */
-static inline void lv_obj_remove_style_all(struct _lv_obj_t * obj)
-{
-    lv_obj_remove_style(obj, NULL, LV_PART_ANY | LV_STATE_ANY);
-}
+void lv_obj_remove_style_all(struct _lv_obj_t * obj);
 
 /**
  * Notify all object if a style is modified

